### PR TITLE
Fix: Free VRAM memory when updating embedding / reranking models

### PR DESCRIPTION
**Pull request by mahenning**
# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

**Before submitting, make sure you've checked the following:**

- [x] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [x] **Description:** Provide a concise description of the changes made in this pull request.
- [x] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [x] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources?
- [x] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
- [x] **Testing:** Have you written and run sufficient tests to validate the changes?
- [x] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [x] **Prefix:** To clearly categorize this pull request, prefix the pull request title using one of the following:
  - **BREAKING CHANGE**: Significant changes that may affect compatibility
  - **build**: Changes that affect the build system or external dependencies
  - **ci**: Changes to our continuous integration processes or workflows
  - **chore**: Refactor, cleanup, or other non-functional code changes
  - **docs**: Documentation update or addition
  - **feat**: Introduces a new feature or enhancement to the codebase
  - **fix**: Bug fix or error correction
  - **i18n**: Internationalization or localization changes
  - **perf**: Performance improvement
  - **refactor**: Code restructuring for better maintainability, readability, or scalability
  - **style**: Changes that do not affect the meaning of the code (white space, formatting, missing semi-colons, etc.)
  - **test**: Adding missing tests or correcting existing tests
  - **WIP**: Work in progress, a temporary label for incomplete or ongoing work

# Changelog Entry

### Description

- This PR aims to fix a dangling VRAM issue when changing/reloading internal (Sentence Transformers) embedding and reranking models. 
- The reranker is now only loaded when **not** Bypass Embedding and Retrieval **and** hybrid search is turned on

### Fixed

- Old models are now properly removed from VRAM
- Fix reranker is always loaded on startup, even if unused

---

### Additional Information

- This PR addresses parts of https://github.com/open-webui/open-webui/discussions/6381
- VRAM freeing is only called when the old model is not externally
- If one switches from internal models on GPU to only external models, ~300-500 MB VRAM are still reserved by torch. This is done by the CUDA driver and can be fixed by restarting the app after switching to external models. Calling `torch.cuda.empty_cache()` won’t drop that.

### Contributor License Agreement

By submitting this pull request, I confirm that I have read and fully agree to the [Contributor License Agreement (CLA)](/CONTRIBUTOR_LICENSE_AGREEMENT), and I am providing my contributions under its terms.

### CODE REVIEW

File: backend/open_webui/routers/retrieval.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import os
 import shutil
 import asyncio
-
+import torch
```
Comment (tjbck) [inline backend/open_webui/routers/retrieval.py]: torch should be imported conditionally

**Comment by tjbck** [comment]
the rest LGTM!

File: backend/open_webui/routers/retrieval.py (hunk 1 above)
Comment (rgaricano) [reply backend/open_webui/routers/retrieval.py]: with DEVICE_TYPE="cuda" or with USE_CUDA="true" ?

Comment (mahenning) [reply backend/open_webui/routers/retrieval.py]: Should be fixed now.

**Comment by tjbck** [comment]
Thanks!
